sdk/logical/events.go

```diff
@@ -30,5 +46,32 @@ type EventType string
 
 // EventSender sends events to the common event bus.
 type EventSender interface {
-	Send(ctx context.Context, eventType EventType, event *EventData) error
+	SendEvent(ctx context.Context, eventType EventType, event *EventData) error
+}
+
+// SendEvent is a convenience method for plugins events to an EventSender, converting the
+// metadataPairs to the EventData structure.
+func SendEvent(ctx context.Context, sender EventSender, eventType string, metadataPairs ...string) error {
+	ev, err := NewEvent()
+	if err != nil {
+		return err
+	}
+	metadata := map[string]string{}
```

Comment:
I assume this change is in progress anyway as I saw the allocations comment was resolved: I think the same algorithm here will work with one less map allocated if we delete this map and use `ev.Metadata.Fields` directly.

Reply:
Oh, oops, let me copy that over.